react.md

```diff
@@ -421,17 +421,289 @@ const SignupSchema = yup.object({
+  <ul className={styles.list}>
+    {items.map(item => (
+      <ListItem 
+        className={styles.item}
+      >    
+        {item}
+      </ListItem>
+    ))}
+  </ul>
+);
+```
+```js
+/* ListItem/index.js */
+import styles from './styles.module.css';
+
+const ListItem = ({ 
+  children, 
+  className = null,
+}) => (
+  <li
+    className={[
+      styles.item,
+      className,
+    ]}
+  >
+    {children}
+  </li>
+);
+```
+
+#### Variables
+If you need compile-time variables, don't jump straight to including Sass. CSS Modules allows declaration of variables that you can import into CSS & and JS modules using its `@value` keyword. This is especially handy for sharing things like common media query breakpoints between your styles and conditional rendering tools like `react-media` (which the below example illustrates) or the DOM API's `window.mediaQueryListener`.
+
+```css
+/* variables.module.css */
+@value colorWhite: #fff;
+@value screenSmall: screen and (min-width: 768px);
+```
+
+```css
+/* MyComponent/styles.module.css */
+@value (
+  colorWhite,
+  screenSmall
+) from 'variables.css';
+
+.container {
+  width: 100%;
+  color: colorWhite;
+}
+
+@media screenSmall {
+  .container {
+    width: 80%;
+  }
+}
+```
+
+```js
+/* MyComponent/index.js */
+import { screenSmall } from 'variables.module.css';
+import SmallScreenComponent from 'MobileFriendlyComponent';
+import LargeScreenComponent from 'MobileFriendlyComponent';
+
+import styles from './styles.module.css';
+
+const MyComponent = ({ 
+  children, 
+}) => (
+  <div className={styles.container}>
+    <Media query={screenSmall}>
+      {matches => (
+        <LargeScreenComponent>
+      ) : (
+        <SmallScreenComponent>
+      )}
+    </Media>
+  </div>
+);
+```
+
+#### Conditional classes
+Install the [`classnames` package](https://www.npmjs.com/package/classnames) if you need to apply classes based on JS data as conditional string concatenation can get unwieldy with two or more optional classes. `classnames`'s array syntax makes this a lot easier to write & read. 
+
+```js
+import classNames from 'classnames';
+
+import styles from './styles.module.css';
+
+const MyComponent = ({ 
+  children,
+  conditionA,
+  conditionB,
+  conditionC,
+}) => (
+  <div 
+    className={classNames([
+      styles.myComponent,
+      conditionA && styles.conditionalClassA,
+      conditionB && styles.conditionalClassB,
+      conditionC && styles.conditionalClassC,
+    ])}
+  >
+    {children}
+  </div>
+);
+```
+
+#### Global selectors
+Only use `:global` selectors to override another library's CSS if there are no other ways of doing so. We should try to make use of third-party UI libraries that allow passing classNames to components for style customization. That said, some otherwise very good tools may not support this or we may need to integrate with non-React JS libraries. Avoid using `:global` otherwise as it opts you out of all the local scoping benefits provided by CSS Modules.
```

Comment:
#SemantiUI case - might be worth mentioning SemanticUI specifically here as a edge case

Reply:
Updated with a SemanticUI example. Great idea!